### DISCUSSION TITLE
Monitor silence from a command

### DISCUSSION THREAD

**Discussion by jdholtz** [kovidgoyal/kitty#6607] [Q&A] [answered]
Does Kitty have a way to monitor when a command produces no output and being able to trigger a terminal bell? An example where this would be useful is be being notified when a process is done or requires user input.

For reference, Konsole and Tmux have this feature (documented [here](https://docs.kde.org/trunk5/en/konsole/konsole/konsole.pdf#page=14) for Konsole and [here](https://man7.org/linux/man-pages/man1/tmux.1.html#OPTIONS) for Tmux (search for `monitor-silence [`)).

I could see this being implemented in a similar way to Konsole as they are both terminal emulators.

**Comment by kovidgoyal** [comment]
https://sw.kovidgoyal.net/kitty/conf/#opt-kitty.tab_activity_symbol

**Reply by jdholtz** [reply]
This appears to work for commands running, but it still stays when a command is asking for input. Here is an example to reproduce:

1. Execute `sleep 2; read -rp test`
2. Switch to another tab before the sleep completes
3. Notice how the tab_activity_symbol doesn't disappear

I believe that the terminals I gave above monitor the output of the command to see if it is active (I didn't look too much into this, so I may be wrong). Then, they allow setting it for an individual window so the user can use it for some commands, but not commands that have no output (such as `cp`). However, it may be possible that Kitty doesn't work in a way where this is possible.

**Comment by kovidgoyal** [comment, marked as answer]
The activity symbold will stay till you switch back to the tab. Its not
a silence symbol its an activity symbol. It indicates something
happened in that tab that you havent seen yet.

**Reply by jdholtz** [reply]
Sounds good. Thank you!

**Comment by kovidgoyal** [comment]
Oh and if what you want is beep when a command finished running simply run it as

command; printf "\a"

or if it is for a command already running, ctrl-z to suspend it, then run fg; print "\a"
